daemon/src/model/cfd.rs

```diff
@@ -2371,6 +2390,212 @@ mod tests {
         assert_eq!(funding_fee.rate, funding_rate);
     }
 
+    #[test]
+    fn given_collab_settlement_then_cannot_start_rollover() {
+        let quantity = Usd::new(dec!(10));
+        let leverage = Leverage::new(2).unwrap();
+        let opening_price = Price::new(dec!(10000)).unwrap();
+
+        let taker_keys = crate::keypair::new(&mut rand::thread_rng());
+        let maker_keys = crate::keypair::new(&mut rand::thread_rng());
+
+        let cfd = Cfd::taker_long()
+            .with_quantity(quantity)
+            .with_opening_price(opening_price)
+            .with_leverage(leverage)
+            .dummy_open(dummy_event_id());
+
+        let _ = assert!(cfd.start_rollover().is_ok());
+
+        let (cfd, _, _, _) = cfd
+            .with_lock(taker_keys, maker_keys)
+            .dummy_collab_settlement_taker(opening_price);
+
+        let err = cfd.start_rollover().unwrap_err();
+        let no_rollover_reason = err.downcast::<NoRolloverReason>().unwrap();
+        assert_eq!(no_rollover_reason, NoRolloverReason::Closed);
```

Comment:
Nit: Could do better use of whitespace here:

```suggestion
	// act
        let (cfd, _, _, _) = cfd
            .with_lock(taker_keys, maker_keys)
            .dummy_collab_settlement_taker(opening_price);
        let result = cfd.start_rollover();
        
        // assert
        let no_rollover_reason = result.unwrap_err().downcast::<NoRolloverReason>().unwrap();
        assert_eq!(no_rollover_reason, NoRolloverReason::Closed);
```